syntax = "proto3";

service Greeter {
  rpc simpleHello (HelloRequest) returns (HelloReply) {}
  rpc serverStreamHello (HelloRequest) returns (stream HelloReply) {}
  rpc clientStreamHello (stream HelloRequest) returns (HelloReply) {}
  rpc biStreamHello (stream HelloRequest) returns (stream HelloReply) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}